server/src/main/java/org/candlepin/async/JobManager.java

```diff
@@ -857,6 +920,16 @@ public synchronized AsyncJobStatus executeJob(final JobMessage message) throws J
             this.setupLogging(status);
             this.setupPrincipal(status);
 
+            // Maybe in this case it'd be better to attempt to use the job key as the job class
+            // rather than failing directly. This would allow use of aliases and explicit job
+            // classes.
```

Comment:
I am not sure how that would work; isn't the job key a string like `EXPORT_JOB`, and not a class? Also, how would using the job key would help here, since it would render pointless the registration mechanism(?) since when `getJobClass` returns a null class, it's because it was not registered.

Reply:
This is related to the e-mail I sent over vacation. I've never been fully sold on the concept of registration and whatnot, and wouldn't mind seeing a system where the job key is either an alias that requires resolution (as it is today) or a job class directly for firing jobs without registering them directly.

At this point it's kind of in-code musings, since it's not high enough priority to really matter.

Reply:
Ah I see. Changing this whole key/registration is fine with me, but I guess not in the scope for this PR.